docs/shadow-props.md

```diff
@@ -208,6 +208,8 @@ export default App;
 </TabItem>
 </Tabs>
 
+---
+
```

Comment:
This change is due to the weird structure the RN website use for many bases, using a `# Reference` heading in the middle of an existing doc that already has a `h1` heading. 

This is not semantically correct, and if it's only to apply a larger font it would be better to apply a different solution.


I've made a little tweak to the heuristic Docusaurus uses to decide (or not) to add the front matter title at the top of the page (https://github.com/facebook/docusaurus/pull/9999), so that I don't have to refactor the other docs as well, but this specific page wasn't covered because it's not consistent with the others and `# Reference` was not prefixed by `---`.

Note other docs are inconsistent:
- Some do not have a `## Example` heading
- Some use a h3 `### Example` heading instead of h2
- Some do not have a `---` before the `# Reference` heading

I only fixed this case because it prevented the insertion of the `# Shadow Prop` heading at the very top of the doc (ie the doc had no title anymore), and will let you figure out further refactors to make RNW more consistent.

Reply:
I agree that we should avoid using an additional `h1`, in other words `# [title]` headers, in pages content. Can make a follow up PR which will cleanup te semantic a bit. Hope that those usage are not blocker for the upgrade.